stream/stream.c

```diff
@@ -861,10 +873,49 @@ char **stream_get_proto_list(void)
             if (*stream_info->protocols[j] == '\0')
                continue;
 
-            MP_TARRAY_APPEND(NULL, list, num,
-                                talloc_strdup(NULL, stream_info->protocols[j]));
+            bool demuxer_match = false;
+            bool lavf_proto = strcmp(stream_info->protocols[j], "ffmpeg") == 0 ||
```

Comment:
I don't understand what you mean.

Reply:
`stream.c` is an abstraction that doesn't concern that it is lavf, ffmpeg or whatever. `stream_info_ffmpeg` and `stream_info_ffmpeg_unsafe` exist and are defined in `stream_lavf.c`. Their protocol list should be populated dynamically if anything.

Reply:
Oh that's what you meant? Is that even desirable? I thought letting those continue to fail at the ffmpeg level was fine personally instead of forcibly rejecting them on the mpv-side.

Reply:
Not sure what you're referring to. This is only about the abstraction.

Reply:
Having the protocols array be dynamic changes the error message on failure when you pass an unsupported protocol. e.g. `mpv smb://fake` becomes:

```
No protocol handler found to open URL smb://
The protocol is either unsupported, or was disabled at compile-time.
```

Instead of
```
[ffmpeg] Protocol not found. Make sure FFmpeg is compiled with networking support.
Failed to open smb://.
```
The latter is a better error message because it makes it clear it's an ffmpeg issue not an mpv one.

Of course, the error message could be changed on our end too.

Reply:
I would worry about the log message so much, both clearly say that protocol is not supported at compile-time. Also note that we maintain list of "some" protocols. What if ffmpeg adds new protocol and it is not added in mpv? By doing this change we know to use always proper list of protocols and early exit if it is not available. If it is only the log message problem, I prefer dynamic protocol init.

Reply:
tbh, I think I would prefer to only keep a whitelist of "safe" protocols for ffmpeg and allow anything else that we get to be populated in the unsafe list.

Reply:
Implemented this.